test/e2e/setup.go

```diff
@@ -423,3 +438,17 @@ var _ = AfterSuite(func() {
 		log.Printf(err.Error())
 	}
 })
+
+var _ = JustAfterEach(func() {
+	gsc := CurrentSpecReport()
+	metricName := strings.Replace(gsc.ContainerHierarchyTexts[0], " ", ".", -1)
+
+	log.WithFields(logrus.Fields{
+		e2emetrics.LogEntryIsE2EEmittableMetric: true,
+		e2emetrics.LogEntryMetricName:           metricName,
+		e2emetrics.LogEntryMetricStatus:         !gsc.Failed(),
```

Comment:
On the bool vs state flag, for example It could be more useful to store the actual state here and decide when the metric is consumed if it is good or bad.